containers-tests/tests/tree-properties.hs

```diff
@@ -101,3 +130,39 @@ prop_monadFix_ls val ta ti =
     f :: (Int -> Int) -> Int -> Tree (Int -> Int)
     f q y = let t = apply ti y
             in fmap (\w -> fact w q) t
+
+prop_toList :: Tree Int -> Property
```

Comment:
I'm sorry I didn't notice this earlier, but in this test suite (and most of the rest in the package) we tend to use `Test.QuickCheck.Poly` where applicable. So for this test we should have

```haskell
prop_toList :: Tree A -> Property
```

The same goes for the rest. Yes, it's a bit less efficient (`A`, `B`, `C`, etc., are actually `Integer`), but clear communication is more important than constant factors for these tests. For `prop_maximum`, ..., `prop_product`, you'll need to use `OrdA`, etc., instead (they have `Ord` and `Num` instances).

Reply:
Ah makes sense, done.